### ISSUE TITLE
Improve git repo age rule

### ISSUE THREAD

**Issue by andys8** [sindresorhus/awesome-lint#105]
I integrated `awesome-lint` for a new project. I keep continuously adding content. Because `awesome-lint` is running on CI and the repository is newly created, the job fails every time. I'd be interested in dead links or relevant warnings but there is no action one could take to solve this. Therefore it's an odd rule.

> Git repository must be at least 30 days old  remark-lint:awesome-git-repo-age

What's the best way to approach it? Can the rule be disabled? Can it be configured to be a warning or be defined as warning in the project?

Thanks for your help :)

**Comment by chrvadala** [comment]
I'm having the same issue and I suspect that it is a firewall issue. 
On my PC lint rule pass, but it fails on github actions https://github.com/chrvadala/awesome-protocols/actions

**Comment by 2color** [comment]
This is because the rule checks the time different against the root hash of the repository. By default, Github Actions will only clone the repo with `--depth 1` which means it has only the latest commit. 

On Github actions, you can define the `fetch-depth: 0` to fetch all commits and for the [linter rule](https://github.com/sindresorhus/awesome-lint/blob/master/rules/git-repo-age.js#L15-L17) to work:

The following Github action should work:

```yaml
name: test
on: push

jobs:
  awesome-lint:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
      with: 
        fetch-depth: 0
    - uses: actions/setup-node@v1
      with:
        node-version: '14.x'
    - run: npm install
    - run: npm test
````

**Comment by chrvadala** [comment]
Thanks you! I was able to fix my workflow https://github.com/chrvadala/awesome-protocols/blob/master/.github/workflows/nodejs.yml

**Comment by joffrey-bion** [comment]
What exactly is the point of this rule?

I thought the idea was to check that the repositories mentioned *in the list* were old enough to be mature, but it looks like it's about the awesome list's repo itself, not the linked repos. Why is that? What's the point?

**Comment by morinted** [comment]
I agree that this rule confuses me a little, because when I set up awesome-lint, it was to ensure that contributions to my list meet the requirements, not that the repo is ready to be PR'd to the main branch.

It feels like maybe a flag to handle switching between the two use cases would be nice… a way to specify that I'd like to look at the contents of readme.md without considering the repository itself.

**Comment by andypiper** [comment]
I've tried adding

`<!--lint disable awesome-git-repo-age-->`

to the `readme.md` file for an Awesome List I just created, but that check is still being run in the CI job and causing the linter to fail. Should I just disable the automation until the repo is past 30 days in age? This is pretty confusing (and it seems a bit "back-to-front", as some of the other comments point out, you'd think it might be more useful for new additions to the list, rather than the list repository itself)

**Comment by nicholaswilde** [comment]
> I've tried adding
> 
> `<!--lint disable awesome-git-repo-age-->`
> 
> to the `readme.md` file for an Awesome List [I just created](https://github.com/andypiper/awesome-modern-twitter-api), but that check is still being run in the CI job and causing the linter to fail. Should I just disable the automation until the repo is past 30 days in age? This is pretty confusing.

I just ran into this same issue. I resolved it by putting the disable line as the first line of the `README.md`.

**Comment by LeoDog896** [comment]
Maybe context can also be provided -- in GitHub, it's a bit hard to check the exact date of commits, so in the lint rule it should maybe say "{x} days left"

**Comment by joffrey-bion** [comment]
Disabling is nice and all, but I'm discussing the very essence of this rule here. I still don't understand why the rule checks the awesome-list repo, instead of checking the repos that are linked to in the list

**Comment by joffrey-bion** [comment]
Thank, now I get it!

**Comment by hlaueriksson** [comment]
Why don't we use the GitHub API to get the repo age?

https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository

The `created_at` property should be whats needed.
